### PR TITLE
Edit defence request from detail

### PR DESCRIPTION
**Pull request by novotnyjakub**
The Submitted defence request show page now has 2 links to "Change this". One for the detainee details and one in the Case details tab.

### CODE REVIEW

**Comment by ansonK** [comment]
After updating Case details the page reloads showing the Interview tab.
Should it not reload with the Case details tab showing (as that was where the User was before) ?

**Comment by novotnyjakub** [comment]
Good point, I'll take a look at that.

**Comment by novotnyjakub** [comment]
OK @ansonK, this should be ready to go. I've added the support for coming back to the same tab. I've also refactored a bit some views and removed the `defence_request_management_spec` for good.

File: app/controllers/defence_requests_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class DefenceRequestsController < BaseController
 
   include DefenceRequestConcern
+  include DefenceRequestsHelper
```
Comment (ansonK) [inline app/controllers/defence_requests_controller.rb]: Do we need to include a view helper into the controller?

Comment (novotnyjakub) [inline app/controllers/defence_requests_controller.rb]: Yeah, so this is something I'm not sure about. Essentially `ApplicationHelper` is automatically available everywhere and I thought we're just splitting it semantically to smaller focused helpers. But if the purpose of all other helpers apart from `ApplicationHelper` is just views, than I'll move my url helper there.

**Comment by ansonK** [comment]
:+1:

**Pull request merged**